### PR TITLE
Adding graphql type generation

### PR DESCRIPTION
**Pull request by SlyCaptainFlint**
Adding graphql type generation. Not actually using the types anywhere, but they are being generated based on the queries/mutations as well as the schema. Types can always be regenerated via `yarn generate-graphql-types`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,20 +20,47 @@
...
+    "@graphql-codegen/typescript": "^1.13.2",
+    "@graphql-codegen/typescript-operations": "^1.13.2"
+  },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "clean": "rimraf build",
+    "generate-graphql-types": "graphql-codegen",
```
Comment (MChartier) [inline package.json:40]: Can we go with the more concise `generate` for now?

File: package.json [hunk 2]
```diff
@@ -20,20 +20,47 @@
...
     "lint": "eslint ./src/ --ext .js --ext .ts --ext .tsx --fix"
   },
   "eslintConfig": {
     "extends": "react-app",
     "rules": {
-      "indent": ["error", 2],
-      "quotes": ["error", "single", { "avoidEscape": true }],
-      "max-len": ["error", { "code": 140, "ignoreTemplateLiterals": true }]
+      "indent": [
```
Comment (MChartier) [inline package.json:46]: I wasn't going to say anything but these rules were a little hypocritical before.

File: codegen.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+schema: http://api-dev.allmy.games/graphql
+documents: ./src/utils/queries.ts
+namingConvention: keep
+generates:
+  ./src/graphql-types.ts:
```
Comment (MChartier) [inline codegen.yml:5]: Can we put this somewhere like `./src/generated/` just to keep generated code separate from authored code?

**Comment by SlyCaptainFlint** [comment]
I'll make updates to use these types in a separate change, as it can be done incrementally

**Pull request merged**